src/Factory/Api/Mastodon/Status.php

```diff
@@ -285,7 +288,13 @@ public function createFromUriId(int $uriId, int $uid = 0, bool $display_quote =
 			$in_reply = [];
 		}
 
-		return new \Friendica\Object\Api\Mastodon\Status($item, $account, $counts, $userAttributes, $sensitive, $application, $mentions, $tags, $card, $attachments, $in_reply, $reshare, $quote, $poll);
+		$queue_count   = $item['delivery_queue_count'];
+		$queue_done    = $item['delivery_queue_done'];
+		$queue_failed  = $item['delivery_queue_failed'];
+		$delivery_data = new FriendicaDeliveryData($queue_count, $queue_done, $queue_failed);
```

Comment:
You could directly pass the fields. And also please ensure that the values are not null.

Reply:
...see new version